tests/unit/test_ecies.cpp

```diff
@@ -0,0 +1,382 @@
+            "D18384",
+            "4D108CEC5BF3E9FE32F5A7F98F49D0C176655DD92CE6A27197F8FA884F6A6F703ECAE253916291E01082540A"
+            "CF22449E",
+            "0A452DBDDD91624E4552B7DA40049251BAD5FA48C1D6D6B4F4499A918B7CFAC391A51006AA862560BBD2CC"
+            "956D3BBC9A50CD758851142C78675E1334EEF796F6",
+            "5E83A26BE96A19856A6E25A102A8794EC1BFED25EA1B1EB4EA188076996BCFAB"
+        },
+        {
+            R"(-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDDVqHt0b/aKqVewnnW9mp80zzPWc93KTveEXsEnyN0Rgj3nheFSdDHL
+z72mLpazSeagBwYFK4EEACKhZANiAASzMIjYvaK0iCR9Snmx2BFFAjaZrCZAMhMh
+0wjM909bwjZt6qd4JJtDc7PDctQQs490bVP8LHUaPbWh4VZHIeX3jz0hKOobpzad
+axVqEbJ5jAX6XNP5AlBLzYHp24XAZwY=
+-----END EC PRIVATE KEY-----
+            )",
+            "0478E3A07F7111C8341A9BBC7FB979F9B1CB27CB0B20E5A0B587C9F43CE1464F53C17461E82F129684D8E6B7B06D74"
+            "AD4AA590985174917E063CEE975E85DA0100626F701EC0278437466B28578852B0EB1684606BEFFF22B9FF6D591C44"
+            "53111A",
+            "1184FC9E16AF814F87DB3CA26F84B408DD876471215F59543D811E6C3E51BA9B0CA446440BBA8A935E99B029"
+            "5708F699",
+            "D29CAB796BC5900C4570B8697F41FED528BA84A339080192E31344FB6F2FF61AF05770BEBA1EA0A0C7A119"
+            "D3C4862D7CE62C4D4176F2178FAAD011A7935A5EAD",
+            "74E0490C9C7E0EE9B4AF03AB44ABDD1479BEC054DDA34A106263535B82B05472"
+        }
+    };
+    return data;
+}
+
+void testECIES(struct testData data)
+{
+    /* Read and set the private key */
+    AsymmetricKeypair privKey(AsymmetricKeypair::readPrivateKeyFromPEM(data.privateKey, ""));
+    ECIESCtxBuilder decBuilder;
+
+    /* Use the default values, which should match here. */
+    auto decCtx = decBuilder.buildDecryptionCtx(privKey);
+
+    /* Generate the ephemeral Key */
+    std::shared_ptr<AsymmetricKey::Spec> keySpec = privKey.getKeySpec();
+    auto eccSpec = std::dynamic_pointer_cast<ECCSpec>(keySpec);
+    ASSERT_TRUE(eccSpec);
+    AsymmetricPublicKey ephKey = AsymmetricPublicKey::fromECPoint(eccSpec, utility::fromHex(data.ephemeralKey));
+
+    /* Set the ephemeral key -> calculate the shared secret */
+    decCtx->setEphemeralKey(ephKey);
+
+    /* Decrypt the ciphertext */
+    decCtx->update(utility::fromHex(data.ciphertext));
+
+    /* Set the received mac value */
+    decCtx->setMAC(utility::fromHex(data.macTag));
+
+    /* Get the plaintext and verify the MAC */
+    auto result = decCtx->finish();
+
+    ASSERT_EQ(result, utility::fromHex(data.plaintext));
+}
+
+class ECIESTest : public testing::TestWithParam<testData> {};
+INSTANTIATE_TEST_CASE_P(ecies,
+                        ECIESTest,
+                        testing::ValuesIn(prepareTestDataForECIESTests()));
+
+TEST_P(ECIESTest, testWithTestVectors)
+{
+    testECIES(GetParam());
+}
+
+TEST(ECIESBuilderTest, incompleteBuilder1)
+{
+    /* Testing for missing parameters. This is the same for Encryption and decryption */
+    ECIESCtxBuilder decBuilder;
+    decBuilder.buildDecryptionCtx(randomKey);
```

Comment:
If this is an incomplete builder, shouldn't this throw?

Reply:
Done